dsp/modules/cache_utils.py

```diff
@@ -24,19 +24,16 @@ def wrapper(*args, **kwargs):
 
 
 cachedir = os.environ.get('DSP_CACHEDIR') or os.path.join(Path.home(), 'cachedir_joblib')
-CacheMemory = Memory(location=cachedir, verbose=0)
+
+CacheMemory = dotdict()
+CacheMemory.cache = noop_decorator
+
+if cache_turn_on:
+    CacheMemory = Memory(location=cachedir, verbose=0)
 
 cachedir2 = os.environ.get('DSP_NOTEBOOK_CACHEDIR')
 NotebookCacheMemory = dotdict()
 NotebookCacheMemory.cache = noop_decorator
 
 if cachedir2:
```

Comment:
Looks like this could be updated to

```
if cache_turn_on and cachedir2:
```

Reply:
It's not obvious what `DSP_NOTEBOOK_CACHEDIR` denotes. Because it seems to assign the `NotebookCacheMemory` a fs cache but then walks that back if `DSP_CACHEBOOL` is true.